src/trace/trace.cpp

```diff
@@ -281,25 +278,80 @@ otf2::definition::system_tree_node& Trace::intern_process_node(pid_t pid)
     }
 }
 
-void Trace::add_process(pid_t pid, pid_t parent, const std::string& name)
+void Trace::add_process(pid_t pid, pid_t parent, const std::vector<std::string>& command_line)
 {
-    auto iname = intern(name);
+    assert(command_line.size() > 0);
+
+    otf2::definition::string prog_name = intern(command_line[0]);
     const auto& parent_node =
         (parent == NO_PARENT_PROCESS_PID) ? system_tree_root_node_ : intern_process_node(parent);
 
     auto emplace_result = system_tree_process_nodes_.emplace(
         std::piecewise_construct, std::forward_as_tuple(pid),
-        std::forward_as_tuple(system_tree_ref(), iname, intern("process"), parent_node));
+        std::forward_as_tuple(system_tree_ref(), prog_name, intern("process"), parent_node));
 
     if (!emplace_result.second) /* emplace failed */
     {
         // process already exists process tree; update its name.
-        update_process_name(pid, iname);
+        update_process_name(pid, prog_name);
         return;
     }
 
     const auto& emplaced_node = emplace_result.first->second;
-    attach_process_location_group(emplaced_node, pid, iname);
+    attach_process_location_group(emplaced_node, pid, prog_name);
+
+    mark_process_begin(pid, prog_name, command_line);
+}
+
+void Trace::mark_process_begin(pid_t pid, const otf2::definition::string& prog_name,
+                               const std::vector<std::string>& command_line)
+{
+    auto now = lo2s::time::now();
```

Comment:
same problem like https://github.com/tud-zih-energy/lo2s/blob/bcf7f78f7c10be11c2c89da31de2075df02fa98e/src/perf/sample/writer.cpp#L2112 but the inverse:
How do you guaranty that now is not actually after the first timestamp delivered from perf? Or worse, what happens when the clock sources differ?